### ISSUE TITLE
Lighthouse Benefits Claims - Implement Validate Endpoint

### ISSUE THREAD

**Issue by sortizsh** [department-of-veterans-affairs/va.gov-team#82149]
## Issue Description
Add support to vets-api to call Lighthouse Benefits Claims - Disability Compensation Claims - Validate endpoint 

---
## Tasks
 - [x] create LH benefits-claims service#validate method
 - [x] implement spec tests for validate method
 
## Acceptance Criteria
 - [x] Method implemented that passes 526 submission data to validate endpoint
 - [x] 100% Passing spec tests
 - [ ] 90% testing line coverage of new code
 - [x] Documentation on how to leverage new service

Breadcrumbs:
- [Benefits Claims Service Implementation](https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/lighthouse/benefits_claims/service.rb) 

- [Disability Compensation Claims Documentation](https://developer.va.gov/explore/api/benefits-claims/docs?version=current)

**Comment by SloopKoning** [comment]
I'm working this ticket in parallel with 83433 since that will be the first consumer of this new service method.

The lighthoust `/validate` endpoint as documented only provides a kind of true or false answer on a 200 response.  All other respone codes are exceptional.  The 200 response provides a `status` attribute that is either "valid" or not.  Therje is no indications that the lighthouse `/validate` endpoint will explicitely say why a 526 submission in not valid.

I will be adding a new instance method to `BenefitsClaims::Service` class named `valid?` which will return either true or false based upon the response received from the lighthouse `/validate` endpoint.  This method will require a ID for a veteran and the from 526 data.  It will look something like this:

```
def valid?(id_of_veteran, form526_data)
...
end
```

w/r/t the 83433 ticket it will be called like this:

```
begin
  BenefitsClaims::Service.new().valid?(id_of_veteran, form_data)
rescue => e
 ...
end
```

**Comment by SloopKoning** [comment]
@tblackwe   a draft PR is available [
](https://github.com/department-of-veterans-affairs/vets-api/pull/16887/files)

**Comment by SloopKoning** [comment]
Specs for automated testing need to be written.

Both Thomas and Scott have had line-by-line walk troughs of the new code.

**Comment by tblackwe** [comment]
In current implementation, the error object is lacking sufficient detail
```
irb(main):004> sub.lighthouse_validation_errors
=> 
[{"title"=>
   "Response Status Code '609' - Common::Exceptions::UnprocessableEntity -- /app/lib/lighthouse/service_exception.rb:37:in `send_error'"}]
```

There is more detail in the response from lighthouse available that we need to log out as well
Please add the detail `description` to the error that is returned.

See staging `FormSubmission.find 1229` for example of failing request 
```
{"description"=>"Account number for the direct deposit.",
 "pattern"=>"^(?:[a-zA-Z0-9]{4,17})?$",
 "type"=>["string", "null"],
 "maxLength"=>14,
 "nullable"=>true,
 "example"=>"123123123123"}
```

**Comment by tblackwe** [comment]
to see the flow,  in staging ArgoCD
```
sub = Form526Submission.find 1229
sub.form_content_valid?
puts sub.lighthouse_validation_errors
```

**Comment by freeheeling** [comment]
@tblackwe The error object you provided has a status code of 609. However, the response status from the /526/validate lighthouse EP for submission with ID 1229 is 422. I'm not sure if what you were providing in both were connected, or you were using the latter as an example. Either way, I'd like to discuss what appears to be a discrepancy between what `lighthouse_validation_errors` returns and the actual response body when submitting to the validate endpoint.

**Comment by freeheeling** [comment]
Submitted draft [PR](https://github.com/department-of-veterans-affairs/vets-api/pull/17077)

**Comment by freeheeling** [comment]
Example result after deploying updates to staging:
```
> sub = Form526Submission.find(1229)
> sub.form_content_valid?
=> false
> pp sub.lighthouse_validation_errors
=> [{"title"=>
   "Response Status Code '422' - Account number for the direct deposit. -- /app/lib/lighthouse/service_exception.rb:37:in `send_error'"}]]
```

**Comment by freeheeling** [comment]
Revisions submitted in [PR](https://app.zenhub.com/workspaces/disability-benefits-experience-team-carbs-6470c8bfffee9809b2634a52/issues/gh/department-of-veterans-affairs/vets-api/17101). Executed script in Staging to confirm updates. Shared with Thomas.

**Comment by freeheeling** [comment]
@tblackwe – A question regarding the "Documentation on how to leverage new service" checkbox in the ticket description's AC – where should this be documented? Or is what was included by Dewayne for those methods he added to the `Form526Submission` model in PR [#16887](https://github.com/department-of-veterans-affairs/vets-api/pull/16887) sufficient?
Thanks!

**Comment by tblackwe** [comment]
What Dewayne put is sufficient. I wasn't sure if the validate script was getting done by same developer, so wanted it documented how to get to it

Thanks for checking

**Comment by freeheeling** [comment]
@tblackwe Suggested by Aurora upon reviewing my PR, and I think it's also a good idea...

> also, this may be out of scope for this, but i was wondering if we can consolidate the two methods:
`validate526` versus `submit526` ...
the `submit526` method already has a mechanism to switch endpoints, defaulting to the synchronous establish claim endpoint.

Wouldn't require much and would significantly streamline code, but not sure where it falls in terms of priority for this or a future sprint. As it is, prospective tickets I've been considered for as assignee are minimal at the moment.
